backend/api/src/api/v1/middleware/auth.rs

```diff
@@ -0,0 +1,62 @@
+use std::sync::{Arc, Weak};
+
+use common::types::session;
+use hyper::http::header;
+use hyper::{Body, StatusCode};
+use routerify::{prelude::RequestExt, Middleware};
+
+use crate::api::error::{ResultExt, RouteError};
+use crate::api::v1::jwt::JwtState;
+use crate::global::GlobalState;
+
+pub fn auth_middleware(_global: &Arc<GlobalState>) -> Middleware<Body, RouteError> {
+    Middleware::pre(|req| async move {
+        let Some(token) = req.headers().get(header::AUTHORIZATION) else {
+            return Ok(req);
+        };
+
+        let global = req
+            .data::<Weak<GlobalState>>()
+            .expect("Global state not found")
+            .upgrade()
+            .ok_or((
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "failed to upgrade global state",
+            ))?;
+        let token = token
+            .to_str()
+            .map_err(|e| (StatusCode::UNAUTHORIZED, "invalid authentication token", e))?;
+
+        // Token's will start with "Bearer " so we need to remove that
+        if !token.starts_with("Bearer ") {
+            return Err(RouteError::from((
+                StatusCode::UNAUTHORIZED,
+                "invalid authentication token",
+            )));
+        }
+
+        let jwt = JwtState::verify(&global, token.trim_start_matches("Bearer "))
```

Comment:
This isnt a match, this is trimming the start of the string. Removing that part of the token. Yes this can be done via a splice, but it doesnt really matter.

Reply:
Hmm, looking at it, `.trim_start_matches()` trims as many matches as it can. It's more idiomatic to use `.strip_prefix()`, it removes exactly once, and returns `Option`, so we don't have to do an extra check (that `if` before this).

Reply:
Seems good you can add it to a subsequent PR.